package/kedro_viz/data_access/repositories/catalog.py

```diff
@@ -26,24 +29,63 @@ def strip_encoding(dataset_name: str) -> str:
 
     @property
     def layers_mapping(self):
-        """Return layer mapping: dataset_full_name -> layer it belongs to in the catalog"""
+        """Return layer mapping: dataset_full_name -> layer it belongs to in the catalog
+        From kedro-datasets 1.3.0 onwards, the 'layers' attribute is defined inside the 'metadata'
+        under 'kedro-viz' plugin.
+
+        Catalog before kedro-datasets 1.3.0:
+            type: pandas.CSVDataSet
+            filepath: /filepath/to/dataset
+            layers: raw
+
+        Catalog from kedro-datasets 1.3.0 onwards:
+            type: pandas.CSVDataSet
+            filepath: /filepath/to/dataset
+            metadata:
+                kedro-viz:
+                    layers: raw
+
+        Currently, Kedro up to 18.x supports both formats. However,
+        support for the old format will be discontinued from Kedro 19.x.
+        Kedro-viz will continue to support both formats.
+        It's recommended to follow the newest format for defining layers in the catalog.
+        """
         if self._layers_mapping is not None:
             return self._layers_mapping
 
-        if self._catalog.layers is None:
-            self._layers_mapping = {
-                self.strip_encoding(dataset_name): None
-                for dataset_name in self._catalog._data_sets
-            }
-        else:
-            self._layers_mapping = {}
-            for layer, dataset_names in self._catalog.layers.items():
-                self._layers_mapping.update(
-                    {
-                        self.strip_encoding(dataset_name): layer
-                        for dataset_name in dataset_names
-                    }
+        self._layers_mapping = {}
+
+        # Maps layers according to the old format
+        if KEDRO_VERSION.match("<=0.19.0"):
```

Comment:
```suggestion
        if KEDRO_VERSION.match("<0.19.0"):
```
I think I made a typo before, sorry... Please do check I got the `match` syntax right also because this was just from memory.

Reply:
it's correct. i checked :)